roles/sp_server_install/tasks/sp_server_install_linux.yml

```
@@ -0,0 +1,111 @@
+- name: Install or Upgrade SP Server
+  block:
+    - name: Delete the existing directory
+      ansible.builtin.file:
+        path: "{{ sp_server_temp_dest }}"
+        state: absent
+
+    - name: Creating required directories for server installation
+      ansible.builtin.file:
+        path: "{{ sp_server_temp_dest }}"
+        state: directory
+        mode: '0755'
+    #  when: sp_server_state == "present"
+
+    - name: Install rsync on remote VMs
+      ansible.builtin.dnf:
+        name: rsync
+        state: present
+
+    - name: Install lsof on remote VMs
+      ansible.builtin.dnf:
+        name: lsof
+        state: present
```

We should ideally use the `ansible.builtin.package` module rather than the dnf module directly. 

Also it makes sense to put these items into a list and perform the install as one:

```suggestion
    - name: Install rsync and lsof
      ansible.builtin.package:
        name: 
          - rsync
          - lsof
        state: present
```